src/main/java/de/hysky/skyblocker/skyblock/SmoothAOTE.java

```diff
@@ -196,20 +206,63 @@ private static TypedActionResult<ItemStack> onItemInteract(PlayerEntity playerEn
         return TypedActionResult.pass(stack);
     }
 
+    /**
+     * Works out if the players location lets them use teleportation or not
+     *
+     * @return if the player should be allowed to teleport
+     */
+    private static boolean isAllowedLocation() {
+        //check mines shafts
+        if (Utils.getMap().equals("Mineshaft")) {
+            return false;
+        } else if (Utils.getIslandArea().equals("⏣ Jungle Temple")) { //do not allow in jungle temple
+            return false;
+        } else if (Utils.getLocation() == Location.PRIVATE_ISLAND && !Utils.getIslandArea().equals("⏣ Your Island")) { //do not allow it when visiting
+            return false;
+        } else if (Utils.getLocation() == Location.PRIVATE_ISLAND && !Utils.getIslandArea().equals("⏣ Your Island")) { //do not allow it when visiting garden
```

Comment:
forget? or I misunderstand something?

Reply:
forget. i was going to do the same check for the garden the realised it was not as easy. oops